bin/quickpkg

```diff
@@ -134,12 +138,36 @@ def quickpkg_atom(options, infos, arg, eout):
 			binpkg_tmpfile = os.path.join(bintree.pkgdir,
 				cpv + ".tbz2." + str(os.getpid()))
 			ensure_dirs(os.path.dirname(binpkg_tmpfile))
-			# The tarfile module will write pax headers holding the
-			# xattrs only if PAX_FORMAT is specified here.
-			tar = tarfile.open(binpkg_tmpfile, "w:bz2",
-				format=tarfile.PAX_FORMAT if xattrs else tarfile.DEFAULT_FORMAT)
-			tar_contents(contents, root, tar, protect=protect, xattrs=xattrs)
-			tar.close()
+			binpkg_compression = settings.get("BINPKG_COMPRESSION", "bzip2")
+			try:
+				compression = _compressors[binpkg_compression]
+			except KeyError as e:
+				eout.error("Invalid or unsupported compression method: %s" % e.args[0])
+				return False
```

Comment:
Nothing checks the quickpkg_atom return value. We want quickpkg to abort an return unsuccessful status in this case.